docs: modify validator system v2 specs (#14)
kroma-network/kroma-specs: specs/experimental/validator-system-v2/validator-manager.md

To become a validator, the first step is to register to the Validator Manager with at least `MIN_REGISTER_AMOUNT` of
KRO tokens. Upon registration, they initiate a [vault](./asset-manager.md#composition-of-asset-manager) via

> self-delegation and set commission rate and dedicated withdraw account. The commission rate is the

Reviewer: ```suggestion
deposit and set commission rate and dedicated withdraw account. The commission rate is the
```

percentage that the validator takes from the output reward. When the validator requests for commission rate change,
he must wait for `COMMISSION_CHANGE_DELAY_SECONDS` to elapse from the moment of requesting commission rate change.

> This safeguard can help delegators choose which validators to delegate to.

Reviewer: ```suggestion
This safeguard can help delegators undelegate before the commission rate change is applied.
```

    bool isInitiated;
    uint8 noSubmissionCount;
    uint8 commissionRate;
    uint8 pendingCommissionRate;
    uint128 commissionChangeInitiatedAt;
}
```

> After registration, validators can receive delegations of KRO and KGH. When total delegated KRO exceeds

Reviewer: ```suggestion
After registration, validators can receive delegations of KRO and KGH. When total deposited and delegated KRO exceeds
```

    bool isInitiated;
    uint8 noSubmissionCount;
    uint8 commissionRate;
    uint8 pendingCommissionRate;
    uint128 commissionChangeInitiatedAt;
}
```

After registration, validators can receive delegations of KRO and KGH. When total deposited and delegated KRO exceeds

> `MIN_ACTIVATE_AMOUNT`, validators can be activated and become eligible to submit output. Note that if the validator

Reviewer: ```suggestion
`MIN_ACTIVATE_AMOUNT`, validators can be activated and become eligible to submit output. Note that if the validators
```

```

As described in [Priority Validator Selection](./overview.md#priority-validator-selection), a validator's weight is

> calculated by summing validator KRO, KRO delegated by KRO delegators, and cumulative reward.

Reviewer: ```suggestion
calculated by summing KRO deposited by validator, KRO delegated by KRO delegators, and cumulative reward.
```

If a validator has not submitted output when it was selected as a priority validator, the validator's
`noSubmissionCount` is incremented by 1. If the validator submits output the next time it is selected as a priority
validator, the `noSubmissionCount` is reset to 0. Otherwise, if the `noSubmissionCount` exceeds `JAIL_THRESHOLD`, the

> validator is deemed to have failed to fulfill its obligations. Accordingly, the validator is sent to jail and removed
> from the validator tree during `SOFT_JAIL_PERIOD`.

Reviewer: ```suggestion
validator is deemed to have failed to fulfill its obligations. Accordingly, the validator is sent to jail during `SOFT_JAIL_PERIOD` and removed
from the validator tree.
```

> Additionally, the validator will be jailed if they submit incorrect outputs or challenge correctly submitted outputs. In

Reviewer: ```suggestion
Additionally, the validator will be jailed if they submit incorrect outputs or challenge incorrectly. In
```

other words, a loser of challenge will result in being jailed for the `HARD_JAIL_PERIOD`. This `HARD_JAIL_PERIOD` is
longer than the `SOFT_JAIL_PERIOD` because such actions are considered threatening to the assets on Layer 2.

After each jail period has elapsed, the validator can be unjailed by calling `tryUnjail` function on its own and
activated if the activation conditions are met.

## Entry Point

Validator Manager contract acts as an entry point for external contracts to slash challenge losers or distribute output
rewards. Below is a detailed description of each case.

> ### Bond

Reviewer: How about move this one to `## Validator Management` section?

> Validator System V2 brings the similar bond mechanism as the previous system. The validator must stake a portion of KRO

Reviewer: ```suggestion
Validator System V2 brings the similar bond mechanism as the previous system. The validator must deposit a portion of KRO
```

deposited as a bond every time they submit an output or create a challenge. Bond amount for each output submission or
challenge creation is defined as `BOND_AMOUNT`. If there is not enough deposited KRO to bond, the validator will no
longer be able to submit outputs or create challenges.

### Slashing

> Slashing occurs during the challenge process. The assets subject to slashing are the bonds used for output submission or

Reviewer: ```suggestion
Slashing occurs during the challenge process. The assets subject to slashing are the bonds bonded during output submission or
```

Reviewer 2: How about this? Because the same words are duplicated in the same sentence.

`The assets subject to slashing are the bonds for output submission`

### Slashing

> Validator System V2 brings the similar bond mechanism as the previous system. The validator must stake a bond every

Reviewer: How about add below contents?
- bond is only reserved using validator KRO, so slashing is only applied to validator
- validator cannot submit an output or create a challenge if they don't have enough deposited KRO to be used as bond

time they submit an output or create a challenge. When the challenge ends, the challenge loser is slashed. Bond amount
for each output submission or challenge creation is defined as `BOND_AMOUNT`.
To apply the slash to the loser, the
[Colosseum](../../fault-proof/challenge.md#contract-interface) contract calls Validator Manager, which calls Asset
Manager to deduct the bond from the validator assets. The loser is sent to jail for`HARD_JAIL_PERIOD`, which is longer
than the `SOFT_JAIL_PERIOD`.

> ### Reward Distribution

Reviewer: Please modify below sentence in this section.

`Otherwise, the rewards for KRO delegation are shared by the validator, KRO delegators, and KGH delegators whose KGH contains KRO`

Reviewer: Also modify this sentence `In addition, any pending challenge rewards accumulated as a result of slash are distributed to the vault of the output submitter who is the challenge winner.` to include the content about tax.

|--------------------------------------------------------------------|-------------------------------------|----------------------------------------|
| $\verb#BASE_REWARD# \times c_i + \verb#BOOSTED_REWARD# \times c_i$ | $\verb#BASE_REWARD# \times (1-c_i)$ | $\verb#BOOSTED_REWARD# \times (1-c_i)$ |

Here, $c_i$ is the commission rate of the validator. Note that a portion of the base reward and the boosted reward is
given as the validator reward through the commission. The remaining boosted rewards are shared among KGH delegators. The
remaining base rewards are shared among KRO delegators and the validator, proportional to the amount of KRO delegated
and deposited.

> In addition, any pending challenge rewards accumulated as a result of slash are given to the validator.

Reviewer: ```suggestion
In addition, any pending challenge rewards accumulated as a result of slash are given to the validator who has won the challenge.
```
